packages/zoe/src/contracts/operaConcertTicket.js

```diff
@@ -0,0 +1,174 @@
+import harden from '@agoric/harden';
+import produceIssuer from '@agoric/ertp';
+import { makeZoeHelpers } from './helpers/zoeHelpers';
+
+/*
+  Roles in the arrangement:
+  - Contract creator: describes the contract with:
+    - number of seats, show, date/time of start
+    - expected (ERTP) amount per ticket (we assume all tickets cost the same)
+  - Smart Contract: 
+    - mints the tickets
+    - provides the seats
+  - Auditorium (unique contract seat, usually taken by the contract creator): the person hosting 
+  the Opera show, selling the tickets and getting the payment back
+  - Ticket buyers (contract seat created on demand): 
+    - can see the available opera show seats
+    - can consult the terms
+    - can redeem the zoe invite with the proper payment to get the ticket back
+
+  ERTP and Zoe are considered to be the most highly trusted pieces of code by everyone
+  They are more trusted than the code of this contract
+  As a consequence, they are going to be leveraged as much as possible by this contract
+  to increase its trustworthiness and by the contract users
+  
+*/
+
+export const makeContract = harden(zoe => {
+  // Create the internal ticket mint
+  const { issuer, mint, amountMath: ticketAmountMath } = produceIssuer(
+    'Opera tickets',
+    'set',
+  );
+
+  const {
+    terms: { show, start, count },
+    issuerKeywordRecord: { Money: moneyIssuer },
+  } = zoe.getInstanceRecord();
+
+  const moneyAmountMath = moneyIssuer.getAmountMath();
+
+  return zoe.addNewIssuer(issuer, 'Ticket').then(() => {
+    // create Zoe helpers after zoe.addNewIssuer because of https://github.com/Agoric/agoric-sdk/issues/802
+    const { rejectOffer } = makeZoeHelpers(zoe);
+
+    // Mint tickets inside the contract
+    // In a more realistic contract, the Auditorium would certainly mint the tickets themselves
+    // but because of a current technical limitation when running the Agoric stack on a blockchain,
+    // minting has to happen inside a Zoe contract https://github.com/Agoric/agoric-sdk/issues/821
+
+    // Mint the tickets ahead-of-time (instead of on-demand)
+    // This way, they can be passed to Zoe + ERTP who will be doing the bookkeeping
+    // of which tickets have been sold and which tickets are still for sale
+    const ticketsAmount = ticketAmountMath.make(
+      harden(
+        Array(count)
+          .fill()
+          .map((_, i) => {
+            const ticketNumber = i + 1;
+            return harden({
+              show,
+              start,
+              number: ticketNumber,
+            });
+          }),
+      ),
+    );
+    const ticketsPayment = mint.mintPayment(ticketsAmount);
+
+    const {
+      invite: contractSelfInvite,
+      inviteHandle: contractOfferHandle,
+    } = zoe.makeInvite();
+    // the contract creates an offer {give: tickets, want: nothing} with the tickets
+    return zoe
+      .getZoeService()
+      .redeem(
+        contractSelfInvite,
```

Comment:
I see what you mean about invite making followed by immediate redemption being annoying unnecessary ceremony. Yes, we should fix this.